tests/Testcontainers.Databases.Tests/DatabasesContainerTest.cs

```diff
@@ -0,0 +1,34 @@
+namespace Testcontainers.Databases;
+
+public sealed class DatabaseContainersTest
+{
+    [Theory]
+    [MemberData(nameof(DatabaseContainersTheoryData))]
+    public void ImplementsIDatabaseContainerInterface(Type type)
+    {
+        Assert.True(type.IsAssignableTo(typeof(IDatabaseContainer)));
+    }
+
+    private static readonly HashSet<Type> NotDatabaseContainerTypes = new()
+    {
+        typeof(LocalStack.LocalStackContainer),
+        typeof(WebDriver.WebDriverContainer),
+    };
+
+    private static bool IsDatabaseContainerType(Type containerType) => !NotDatabaseContainerTypes.Contains(containerType);
+
+    public static IEnumerable<object[]> DatabaseContainersTheoryData
+    {
+        get
+        {
+            static bool HasGetConnectionStringMethod(Type type) => type.IsAssignableTo(typeof(IContainer)) && type.GetMethod("GetConnectionString") != null;
+            var assembly = typeof(DatabaseContainersTest).Assembly;
+            var dependencyContext = DependencyContext.Load(assembly) ?? throw new InvalidOperationException($"DependencyContext.Load({assembly}) returned null");
+            return dependencyContext.RuntimeLibraries
+                .Where(library => library.Name.StartsWith("Testcontainers."))
+                .SelectMany(library => Assembly.Load(library.Name).GetExportedTypes().Where(HasGetConnectionStringMethod))
+                .Where(IsDatabaseContainerType)
+                .Select(type => new[] { type });
```

Comment:
Can you share why `DependencyContext` is necessary? What about the following (this makes `Microsoft.Extensions.DependencyModel` unnecessary:

```suggestion
            return Directory.GetFiles(".", "Testcontainers.*.dll", SearchOption.TopDirectoryOnly)
                .Select(Path.GetFullPath)
                .Select(Assembly.LoadFrom)
                .SelectMany(assembly => assembly.GetExportedTypes())
                .Where(type => type.IsAssignableTo(typeof(IContainer)))
                .Where(type => type.GetMethod("GetConnectionString") != null)
                .Where(IsDatabaseContainerType)
                .Select(type => new []{ type });
```

Reply:
You are right, `DependencyContext` was not necessary. The test was rewritten a bit differently in 22f383c84db08b99879468682abd2f8c3f726590 but I have removed the `Microsoft.Extensions.DependencyModel` dependency anyway.